src/Dialect/Mlir/IndexExprDetail.cpp

```diff
@@ -55,9 +50,18 @@ void IndexExprImpl::initAsUndefined() {
 }
 
 void IndexExprImpl::initAsQuestionmark() {
-  const std::lock_guard<std::mutex> lock(indexExprQuestionMarkMutex);
+  // Question mark has value of -1 by default.
+  init(/*isDefined*/ true, /*literal*/ false, IndexExprKind::Questionmark, -1,
+      AffineExpr(nullptr), Value(nullptr));
+}
+
+void IndexExprImpl::initAsQuestionmark(Value val, int64_t axis) {
+  // Each question mark is assigned a unique integer that is obtained
+  // by hashing the tensor value and the target axis/dimension.
+  llvm::hash_code questionValue =
```

Comment:
Can you just indicate in the code what is the scope of this hash? And where it is declared, tx.